### PR TITLE
Add Settings screen

### PR DESCRIPTION
**Pull request by grittaweisheit**
Add settings screen

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -98,4 +98,14 @@ dependencies {
...
 
-}
\ No newline at end of file
+    // DataStore
+    implementation("androidx.datastore:datastore-preferences:1.0.0")
+    //For Coroutines if not added earlier
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.4'
+    //For lifecycle mybe unnecessary
+    implementation "androidx.lifecycle:lifecycle-runtime-ktx:2.6.0-alpha03"
+    implementation "androidx.lifecycle:lifecycle-livedata-ktx:2.6.0-alpha03"
```
Comment (MichaelBuessemeyer) [inline app/build.gradle:107]: We already have these dependencies in other versions (and not alpha) 😅 

I would delete adding these or the once already present

File: app/src/main/java/com/example/mytrainingpal/components/CustomTimeInput.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun CustomTimeInput(
+    value: String,
+    onValueChange: (String) -> Unit,
+    postText: String? = null,
+    backgroundColor: Color = MaterialTheme.colors.background
+) {
+    // https://medium.com/@daniel.atitienei/date-and-time-pickers-in-jetpack-compose-f641b1d72dd5
```
Comment (lenilol) [inline app/src/main/java/com/example/mytrainingpal/components/CustomTimeInput.kt:33]:
```suggestion
    // usage learned from https://medium.com/@daniel.atitienei/date-and-time-pickers-in-jetpack-compose-f641b1d72dd5
```

File: app/src/main/java/com/example/mytrainingpal/components/CustomTimeInput.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        { _, selectedHour: Int, selectedMinute: Int ->
+            selectedTimeText.value = listOf<Int>(selectedHour, selectedMinute).joinToString(":") {
+                it.toString().padStart(2, '0')
+            }
+            onValueChange(selectedTimeText.value)
+        }, hour, minute, true
+    )
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(0.dp),
```
Comment (lenilol) [inline app/src/main/java/com/example/mytrainingpal/components/CustomTimeInput.kt:51]: 0.dp ?

File: app/src/main/java/com/example/mytrainingpal/prefrences/PreferencesDataStoreHelper.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import java.io.IOException
+
+// see https://proandroiddev.com/preference-datastore-the-generic-way-d26b11f1075f
```
Comment (lenilol) [inline app/src/main/java/com/example/mytrainingpal/prefrences/PreferencesDataStoreHelper.kt:15]:
```suggestion
// inspired by https://proandroiddev.com/preference-datastore-the-generic-way-d26b11f1075f
```

File: app/src/main/java/com/example/mytrainingpal/screens/Settings.kt [hunk 5]
```diff
@@ -1,11 +1,234 @@
...
+            (preferences[PreferencesConstants.NOTIFICATION_TIME_KEY.name] ?: "12:00") as String
+        defaultBreak = (preferences[PreferencesConstants.DEFAULT_BREAK_KEY.name] ?: 30) as Int
+        preferences
+    }
+
+    TabScreen(
+        tabContent = {
+            MainSettingsScreenContent(userName = userName,
+                updateName = {
+                    //preferencesViewModel.setName(it)
```
Comment (lenilol) [inline app/src/main/java/com/example/mytrainingpal/screens/Settings.kt:73]: Just a leftover ???

**Pull request merged**